### PR TITLE
Putting bluefin daily volume DEX adaptor

### PR DESCRIPTION
**Pull request by singhalhemant**
Putting bluefin daily volume DEX adaptor

### CODE REVIEW

**Comment by llamatester** [comment]
The bluefin adapter exports: 
        
 
 _______________________________________
    Dexs for 10/10/2023
    _______________________________________
    
    Version -> SUI
    ---------
    SUI 👇
    Backfill start time: 27/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 7104326.151
    Timestamp: 1696982399
    
    
    Version -> ARB
    ---------
    ARBITRUM 👇
    Backfill start time: 27/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 6252271.016600002
    Timestamp: 1696982399
    
    
    Done in 6.11s.

**Comment by llamatester** [comment]
The bluefin adapter exports: 
        
 
 _______________________________________
    Dexs for 10/10/2023
    _______________________________________
    
    Version -> SUI
    ---------
    SUI 👇
    Backfill start time: 27/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 8120550.3582
    Timestamp: 1696982399
    
    
    Version -> ARB
    ---------
    ARBITRUM 👇
    Backfill start time: 27/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 7058168.5669
    Timestamp: 1696982399
    
    
    Done in 5.92s.

**Comment by singhalhemant** [comment]
Moved to using marketdata API

File: dexs/bluefin/index.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  "sui": "https://dapi.api.sui-prod.bluefin.io/marketData",
+}
+
+const product_config = {
+  "arb": ["ETH-PERP", "BTC-PERP"],
+  "sui": ["ETH-PERP", "BTC-PERP"],
+}
+
+const fetchURL = (baseURL: string, product: string) => {
+  baseURL = `${baseURL}` + `?symbol=${product}`;
```
Comment (YameenMalik) [inline dexs/bluefin/index.ts]: please change this to `${baseURL}?symbol=${product}`;

File: dexs/bluefin/index.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  return baseURL;
+}
+
+const computeVolume = async (timestamp: number, baseUrl: string, productIds: string[]) => {
```
Comment (YameenMalik) [inline dexs/bluefin/index.ts]: Please define a return type for the method.  It is best to create an interface at the top for the return type:
```
return {
    totalVolume: undefined,
    dailyVolume: dailyVolume ? `${dailyVolume}` : undefined,
    timestamp: timestamp,
  };
  ```
  Something like:
  ```
  interface Volume {
     totalVolume: number | undefined,
     dailyVolume: number | undefined,
     timestamp: number,     
  }
  ```

File: dexs/bluefin/index.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+const fetchSUI = async (timeStamp: number) => {
+  return await computeVolume(timeStamp, url_config.sui, product_config.sui);
+};
+
+const fetchArb = async (timeStamp: number) => {
+  return await computeVolume(timeStamp, url_config.arb, product_config.arb);
+};
+
+const startTime = 1695857191;
```
Comment (YameenMalik) [inline dexs/bluefin/index.ts]: define constant in `CAPITAL_CASE`

Comment (YameenMalik) [reply dexs/bluefin/index.ts]: what is this time by the way? would be good to add a comment. I assume its the time at which trading on Sui began.

**Review by YameenMalik — CHANGES_REQUESTED** [review]

**Comment by llamatester** [comment]
The bluefin adapter exports: 
        
 
 _______________________________________
    Dexs for 10/10/2023
    _______________________________________
    
    Version -> SUI
    ---------
    SUI 👇
    Backfill start time: 27/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 8081431.164000001
    Timestamp: 1696982399
    
    
    Version -> ARB
    ---------
    ARBITRUM 👇
    Backfill start time: 27/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 7053238.391799999
    Timestamp: 1696982399
    
    
    Done in 5.12s.

**Comment by llamatester** [comment]
The bluefin adapter exports: 
        
 
 _______________________________________
    Dexs for 10/10/2023
    _______________________________________
    
    Version -> SUI
    ---------
    SUI 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 8073534.4944
    Timestamp: 1696982399
    
    
    Version -> ARB
    ---------
    ARBITRUM 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 7046081.088399999
    Timestamp: 1696982399
    
    
    Done in 5.40s.

**Comment by llamatester** [comment]
The bluefin adapter exports: 
        
 
 _______________________________________
    Dexs for 10/10/2023
    _______________________________________
    
    Version -> SUI
    ---------
    SUI 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 8075989.1836
    Timestamp: 1696982399
    
    
    Version -> ARB
    ---------
    ARBITRUM 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 7046142.5101
    Timestamp: 1696982399
    
    
    Done in 4.76s.

**Review by YameenMalik — APPROVED** [review]

**Comment by llamatester** [comment]
The bluefin adapter exports: 
        
 
 _______________________________________
    Dexs for 11/10/2023
    _______________________________________
    
    Version -> SWAP
    ---------
    ARBITRUM 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 4120592.088
    Timestamp: 1697068799
    
    
    SUI 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 7231390.791300001
    Timestamp: 1697068799
    
    
    Version -> DERIVATIVES
    ---------
    ARBITRUM 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 4120592.088
    Timestamp: 1697068799
    
    
    SUI 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 7231390.791300001
    Timestamp: 1697068799
    
    
    Done in 5.46s.

**Comment by llamatester** [comment]
The bluefin adapter exports: 
        
 
 _______________________________________
    Dexs for 11/10/2023
    _______________________________________
    
    Version -> SWAP
    ---------
    ARBITRUM 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 4538568.897
    Timestamp: 1697068799
    
    
    SUI 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 7396416.920700001
    Timestamp: 1697068799
    
    
    Done in 4.23s.

**Comment by dtmkeng** [comment]
seem symbol your config is perp is it DERIVATIVES volume ? 
not swap, swap mean spot or swap direct to LP ?

**Comment by singhalhemant** [comment]
Hi @dtmkeng , 

Thanks for your comment. We need our volume 24h to be available in this dashboard as well https://defillama.com/chain/Sui (against bluefin). 

Will it be ok if I move the config to derivatives? 

We are a trading exchange which supports perp for now but have plans to add spots soon as well.

**Comment by dtmkeng** [comment]
> Hi @dtmkeng ,
> 
> Thanks for your comment. We need our volume 24h to be available in this dashboard as well https://defillama.com/chain/Sui (against bluefin).
> 
> Will it be visible in "Volume 24h" column if I move the config to derivatives?
> 
> We are a trading exchange which supports perp for now but have plans to add spots soon as well.
this page  volume volume 24h I guess supply swap volume 
if you need list DERIVATIVES volume it show only this page https://defillama.com/derivatives
and toggle like this 
image

**Comment by llamatester** [comment]
The bluefin adapter exports: 
        
 
 _______________________________________
    Dexs for 11/10/2023
    _______________________________________
    
    Version -> SWAP
    ---------
    ARBITRUM 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 4609461.236200001
    Timestamp: 1697068799
    
    
    SUI 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 3543628.4166
    Timestamp: 1697068799
    
    
    Version -> DERIVATIVES
    ---------
    ARBITRUM 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 4609461.236200001
    Timestamp: 1697068799
    
    
    SUI 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 3543628.4166
    Timestamp: 1697068799
    
    
    Done in 5.80s.

**Comment by singhalhemant** [comment]
Hi @dtmkeng ,

Thanks for your guidance. Checked and it would be good to have it in both given we are a trading protocol. 

Made this quick change and will really appreciate if you could please approve and merge! 

"this page volume volume 24h I guess supply swap volume
if you need list DERIVATIVES volume it show only this page https://defillama.com/derivatives
and toggle like this"

**Comment by dtmkeng** [comment]
> Hi @dtmkeng ,
> 
> Thanks for your guidance. Checked and it would be good to have it in both given we are a trading protocol.
> 
> Made this quick change and will really appreciate if you could please approve and merge!
> 
> "this page volume volume 24h I guess supply swap volume if you need list DERIVATIVES volume it show only this page https://defillama.com/derivatives and toggle like this"

your API is have swap volume ?

**Comment by dtmkeng** [comment]
seem `PRODUCT_CONFIG` is perp

**Comment by dtmkeng** [comment]
and if you have one type volume plx add only one breakdown

**Comment by llamatester** [comment]
The bluefin adapter exports: 
        
 
 _______________________________________
    Dexs for 13/10/2023
    _______________________________________
    
    Version -> DERIVATIVES
    ---------
    ARBITRUM 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 4224060.9303
    Timestamp: 1697241599
    
    
    SUI 👇
    Backfill start time: 25/9/2023
    NO METHODOLOGY SPECIFIED
    Total volume: undefined
    Daily volume: 1805908.102
    Timestamp: 1697241599
    
    
    Done in 4.84s.

**Comment by singhalhemant** [comment]
Hi @dtmkeng , 

Yes sure. For now, I have removed swap and going with derivatives so that this PR could be merged. 

Will really appreciate if it can be merged now please.

**Pull request merged**